src/Google.Api.Gax.Grpc/CallSettings.cs

```diff
@@ -35,14 +35,39 @@ public sealed class CallSettings
             CallTiming timing,
             Action<Metadata> headerMutation,
             WriteOptions writeOptions,
-            ContextPropagationToken propagationToken)
+            ContextPropagationToken propagationToken) : this(cancellationToken, credentials, timing, headerMutation, writeOptions, propagationToken, null, null)
+        {
+        }
+
+        /// <summary>
+        /// Constructs an instance with the specified settings.
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation token that can be used for cancelling the call.</param>
+        /// <param name="credentials">Credentials to use for the call.</param>
+        /// <param name="timing"><see cref="CallTiming"/> to use, or null for default retry/expiration behavior.</param>
+        /// <param name="headerMutation">Action to modify the headers to send at the beginning of the call.</param>
+        /// <param name="writeOptions"><see cref="global::Grpc.Core.WriteOptions"/> that will be used for the call.</param>
+        /// <param name="propagationToken"><see cref="ContextPropagationToken"/> for propagating settings from a parent call.</param>
+        /// <param name="responseMetadataHandler">Action to invoke when response metadata is received.</param>
+        /// <param name="trailingMetadataHandler">Action to invoke when trailing metadata is received.</param>
+        public CallSettings(
+            CancellationToken? cancellationToken,
+            CallCredentials credentials,
+            CallTiming timing,
+            Action<Metadata> headerMutation,
+            WriteOptions writeOptions,
+            ContextPropagationToken propagationToken,
+            Action<Metadata> responseMetadataHandler,
+            Action<Metadata> trailingMetadataHandler)
```

Comment:
We should consider allowing async handlers: `Func<Metadata, Task> ...Handler`

Reply:
Hmm, possibly. That does make it all more complex - and I'd argue that you almost certainly don't want to be doing significant work in the handler anyway.